toonz/sources/toonz/ocaio.cpp

```diff
@@ -534,4 +535,379 @@ void ExportOCACommand::execute() {
+        m_dpi);
+    TXshSimpleLevel *sl = dynamic_cast<TXshSimpleLevel *>(level);
+    // we filmmakers don't use dpi ??
+    sl->getProperties()->setDpi(m_dpi);
+    sl->getProperties()->setDpiPolicy( LevelProperties::DP_CustomDpi);
+    sl->getProperties()->setWhiteTransp(m_whiteTransp);
+    sl->getProperties()->setDoPremultiply(m_doPremultiply);
+    sl->getProperties()->setDoAntialias(m_antialiasSoftness);
+
+    // float colorSpaceGamma   = 2.2;
+    // we don't need to touch this...
+    // display gamma 2.2 is only valid for srgb color space, and for rec709 is
+    // grading on a computer monitor...but... gamma for rec709 should be 2.4 if
+    // you have a rec709 grading monitor set for HDTV, gamma for rec2020 should
+    // be 2.6 if you have a rec2020 grading monitor, gamma for DCI-P3 should
+    // be 2.4 if you have a DCI-P3 grading theatre !? linear images colorDepth
+    // could be could be F16 or F32 (but the channel
+    // width preference is locked to 32 bit if checking linear color space...) 
+    //if (sl->getPath().getType() == "exr")
+    //    sl->getProperties()->setColorSpaceGamma(colorSpaceGamma); 
+    // Why is colorSpaceGamma disabled in the level settings ui ??
+    // see render>output settings> color settings
+
+    sl->setName(jsonLayer["name"].toString().toStdWString());
+    std::vector<TFrameId> fids;
+    for (auto frame : jsonLayer["frames"].toArray()) {
+      importOcaFrame(frame.toObject(), sl);
+      TFrameId fid(frame.toObject()["frameNumber"].toInt());
+      fids.push_back(fid);
+    }
+    
+    // A comment about the current api : exposeLevel should be fixed to use fids :
+    // insert column just creates an empty column :
+    // m_xsheet->insertColumn(m_xsheet->getColumnCount(), TXshColumn::ColumnType::eLevelType);
+    // But how to fill the xsheet column ? looking at :
+    // Looking in LoadLevelPopup::execute().
+    // This works but exposes in 1's :
+    // IoCmd::exposeLevel(sl, 0, m_xsheet->getColumnCount(), true, true); 
+    // ok but, strangely, even the following methods, with fids argument, don't expose with correct timing !  (exposes in 1's too) :
+    // fids are only used for source time, not destination time (even when isAnimationSheetEnabled is true)
+    // IoCmd::exposeLevel( sl, 0, m_xsheet->getColumnCount(), fids, true, true);
+    // IoCmd::exposeLevel( sl, 0, m_xsheet->getColumnCount(), fids);
+    // m_xsheet->exposeLevel(0, m_xsheet->getFirstFreeColumnIndex(), sl, fids, true);
+    // I think this method should be fixed to use the fids for destination time in the column
+    // (eventually adding an optional bool argument timingUseFids,
+    // but i don't get why we pass fids if we're not using those ??)...
+
+    // that's the very simple solution...
+    int emptyColumnIndex = m_xsheet->getFirstFreeColumnIndex();
+    for (auto fid : fids)
+      m_xsheet->setCell(fid.getNumber(), emptyColumnIndex, TXshCell(sl, fid));
```

Comment:
Fid has 2 meanings which will change if you are in Level Edit mode (Level Strip) or Scene Edit mode (Xsheet)

In Level Edit mode, the Fid is the frame number as it appears in the level strip.  This usually begins with 1.  It's rare that it starts with 0.

In Scene mode, the Fid is the Scene Frame position, which starts at 0 (internally).  For placement in the xsheet, you need to do `fid.getNumber() - 1`.  If Frame 1 is the beginning of the animation, then internally the 1st frame is 0.  No doing this would cause the animation to start on Scene Frame 2.  The only time this may be a problem is if the OCA data somehow has `"frameNumber": 0` since there isn't a -1 Scene Frame and will crash when trying to load

Reply:
I also am offseting any negative frame range now...